src/main/kotlin/io/sip3/salto/ce/sdp/SdpHandler.kt

```diff
@@ -56,9 +59,17 @@ class SdpHandler : AbstractVerticle() {
         vertx.eventBus().localConsumer<SipTransaction>(RoutesCE.sdp + "_session") { event ->
             try {
                 val transaction = event.body()
-                handle(transaction)
+
+                handle(transaction) { asr ->
```

Comment:
Implementation here looks too complicated. In fact what we need is a simple method which will return `SdpSession` object (without `completionHandler` and other async related stuff). Also, I think we can simply keep all the exceptions related stuff inside SdpHandler itself and do not expose it further:
```
                try {
                    val session = handle(transaction) 
                } catch (e: Exception) {
                    logger.error("SdpHandler 'handle()' failed.", e)
                    event.reply(null)
                }
```